core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -107,7 +108,7 @@ private String metadataFileLocation(Table table) {
 
   private FileIO fileIO(Table table) {
     if (table.io() instanceof HadoopConfigurable) {
-      ((HadoopConfigurable) table.io()).serializeConfWith(conf -> new SerializableConfiguration(conf)::get);
+      ((HadoopConfigurable) table.io()).serializeConfWith(SerializableConfSupplier::new);
```

Comment:
I’m surprised we don’t have to update more code to call `.get` now. I remember that the reason for adding `::get`to `serializeConfWith` was to avoid users needing to call `.get` depending on the place.

Since things are all passing, I’m guessing not?

Reply:
What's happening is that the lambda is creating a `SerializableConfiguration` and returning its `get` method as a lambda that implements `SerializableSupplier`. In the new version, the lambda is removed and the returned object is a `SerializableSupplier`.